### DISCUSSION TITLE
Getting the value for field _IsRecord

### DISCUSSION THREAD

**Discussion by sabrinadeveloper2023** [pnp/pnpcore#1157] [Q&A]
Hi,

I am implementing a function to delete a file in a document library. Before I delete the file I need to check whether the file is a Record or not, therefore I need to check the value in the field "_IsRecord". This field is part of the SharePoint Record Management functionality.

My question is how I can get the value in this field? (Values can be Yes or No)

This is my code:

            using (var context = await _pnpContextFactory.CreateAsync(siteUrl))
            {
            // Get a reference to the file
                  IFile retrievedDocument = await context.Web.GetFileByIdAsync(deleteDocument.ID,
                                f => f.ListItemAllFields.QueryProperties(li => li.All,
                                    li => li.ParentList.QueryProperties(p => p.Title,
                                        p => p.Fields.QueryProperties(p => p.InternalName, p => p.FieldTypeKind,
                                                                      p => p.TypeAsString, p => p.Title))));
            
                  var retrievedList = retrievedDocument.ListItemAllFields.ParentList;
            
                  var fieldIsRecord = await retrievedList.Fields.Where(p => p.InternalName == "_IsRecord").FirstOrDefaultAsync();
            
                  // -- Add condition here to check if field IsRecord is set to Yes
            }

Thank you for your help!

**Comment by sabrinadeveloper2023** [comment]
Below the schema for the _IsRecord field
![image](https://user-images.githubusercontent.com/130606153/231576963-7b45291c-91d9-4fc6-8e30-8c9e0fcfe5de.png)

**Comment by sabrinadeveloper2023** [comment]
Hi,

I noticed the following sentence is not returning the field:

  var fieldIsRecord = await retrievedList.Fields.Where(p => p.InternalName == "_IsRecord").FirstOrDefaultAsync();

This is what I need:

I get a request to delete a file and with that request I only get the SiteUrl and the File Guid. I need to do the following:

1-Check the value in field _IsRecord, this field is part of the listitem collection and it is not part of the ListItemAllFields, I assume I need the reference to the item in the list. How I can get the item from the File Guid?
2- Then I need to check the value in field _IsRecord.
3-If value is No, I proceed to delete the file.

Thank you very much for your time and help, much appreciated

**Comment by sabrinadeveloper2023** [comment]
I think I can get the field as follow, but I am not sure how to get the value store in this type of field.

Thanks

 // Get a reference to the file
                    IFile retrievedDocument = await context.Web.GetFileByIdAsync(deleteDocument.ID,
                                  f => f.ListItemAllFields.QueryProperties(li => li.All,
                                      li => li.ParentList.QueryProperties(p => p.Title,
                                          p => p.Fields.QueryProperties(p => p.InternalName, p => p.FieldTypeKind, p => p.TextField,
                                                                        p => p.TypeAsString, p => p.Title))));

                    IField field = await retrievedDocument.ListItemAllFields.ParentList.Fields.Where(f => f.InternalName == "_IsRecord").FirstOrDefaultAsync();

                    if (field != null)
                    {
                        // -- Add condition here to check if field IsRecord is set to Yes
                    }

**Comment by jansenbe** [comment]
See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core.Test/SharePoint/ListItemTests.cs#L5047-L5095 for a sample on compliance work, it also shows how to retrieve those fields via a CAML query.
